Add Storage Destination Limitation with Insert Functions (#6404)
segmentio/segment-docs: src/connections/functions/insert-functions.md

You can also use this page to [enable destination insert functions](#enable-the-insert-function) in your workspace.

> > warning "Storage Destination Limit"

Reviewer: ```suggestion
> warning "Storage destinations are not compatible with Destination Insert Functions"
```

> > Currently, you are not able to connect a Storage Destination to an Insert Function.

Reviewer: hi @courtneyga - do you have more context on why this is? Other places in the docs where one product is incompatible with another we mention the "why" behind it and I'd like to include that here too!

> > Currently, you are not able to connect a Storage Destination to an Insert Function.

Reviewer: ```suggestion
> You cannot connect an Insert Function to a storage destination at this time.
```
